packages/langchain/src/openai/types.ts

```diff
@@ -0,0 +1,55 @@
+import type { BaseChatModelCallOptions, BaseChatModelParams } from '@langchain/core/language_models/chat_models';
+import { BaseLLMParams } from '@langchain/core/language_models/llms';
+import type { OpenAiChatCompletionParameters } from '@sap-ai-sdk/foundation-models';
+import type {
+  AzureOpenAiChatModel,
+  CustomRequestConfig
+} from '@sap-ai-sdk/core';
+import type { ModelConfiguration, ResourceGroupConfiguration } from '@sap-ai-sdk/ai-api';
+
+/**
+ * Input type for OpenAI chat models.
+ */
+export type OpenAiChatModelParams = Omit<OpenAiChatCompletionParameters, 'messages'> &
+  BaseChatModelParams &
+  ModelConfiguration<AzureOpenAiChatModel> &
+  ResourceGroupConfiguration;
+
+/**
+ * Chat model call options for OpenAI.
+ */
+export type OpenAiChatCallOptions = CustomRequestConfig & BaseChatModelCallOptions;
+
+/**
+ * Input type for OpenAI embedding models.
+ */
+export type OpenAiEmbeddingModelParams = ModelConfiguration<AzureOpenAiChatModel> &
+  ResourceGroupConfiguration &
+  BaseLLMParams;
+
+/**
+ * OpenAI toolchoice type.
+ */
+export type ToolChoice =
```

Comment:
[q] does this need to be public? Seems to me that it should be internal. Or maybe even moved to where is is used and never exported at all. Also applies to `LangChainToolChoice`. Consider whether other types should be hidden as well.